src/AutoRest.CSharp/Common/Output/Models/Types/ModelTypeProvider.cs

```diff
@@ -134,12 +157,12 @@ private IEnumerable<JsonPropertySerialization> CreatePropertySerializations()
             {
                 foreach (var property in objType.Properties)
                 {
-                    if (property.InputModelProperty is null)
+                    if (property.InputModelProperty is not { } inputModelProperty)
                         continue;
 
                     var declaredName = property.Declaration.Name;
                     var serializedName = property.InputModelProperty.SerializedName ?? property.InputModelProperty.Name;
-                    var optionalViaNullability = !property.IsRequired && !property.ValueType.IsNullable && !TypeFactory.IsCollectionType(property.ValueType);
+                    var optionalViaNullability = !inputModelProperty.IsRequired && !inputModelProperty.Type.IsNullable && !TypeFactory.IsCollectionType(property.ValueType);
```

Comment:
What's the difference between `property.IsRequired` and `property.InputModelProperty.IsRequired`?

Reply:
They are the same. I changed it because of alignment (mgmt is doing this in this way)